Add `go-guru` support

sdevlin opened the pull request:
This is to add support for identifier highlights in `go-guru` as per https://github.com/bbatsov/zenburn-emacs/issues/282.

I have no strong opinion on the colors. I just picked something I thought would be noticeable but not too garish. I'm happy to change it.

Also, I'm not sure whether it's recommended to continue to inherit from the `highlight` face, but that's what I did. I'm happy to change this as well.

Before:

<img width="580" alt="before" src="https://user-images.githubusercontent.com/174996/30089384-3575c832-9261-11e7-8fbc-4986d8980c00.png">

After:

<img width="580" alt="after" src="https://user-images.githubusercontent.com/174996/30089395-44cead58-9261-11e7-948f-34163df3530f.png">

basil-conto commented:
> I'm not sure whether it's recommended to continue to inherit from the `highlight` face

Seeing as you're overriding any and all default and `zenburn`-assigned attributes of the `highlight` face, I'd argue the inheritance is unnecessary.

> I have no strong opinion on the colors

Same, especially since I don't use the mode (yet :). One way to avoid having to pick colours, however, would be to use a built-in face which serves a similar purpose, e.g. `isearch`, `lazy-highlight` or `match` (I'm only suggesting this as a general tip, not asking you to pick one of these instead of the current colours).

sdevlin commented:
Thanks @basil-conto!

> Seeing as you're overriding any and all default and `zenburn`-assigned attributes of the `highlight` face, I'd argue the inheritance is unnecessary.

Okay, removed. I rebased to keep the PR clean; I hope that's okay.

> One way to avoid having to pick colours, however, would be to use a built-in face which serves a similar purpose, e.g. `isearch`, `lazy-highlight` or `match`

I experimented with some of these faces, but they didn't feel quite right to me. To my mind, this feature is more in the spirit of `hi-lock`, so I made a face similar to the `hi-*` faces but with a more muted background color.

Of course, I'm willing to defer to anyone with more UI expertise.

Let me know if you need anything else!
